Add pdf.spans.manual

ines opened the pull request:
Requires the latest Prodigy v1.17.1.

**Docs preview:** https://deploy-preview-294--prodi-gy.netlify.app/docs/plugins#pdf-spans.manual

ines committed:
Add pdf.spans.manual

ines committed:
Update types for consistency

ines committed:
Only add image span if bounding box is available

magdaaniol committed:
fix entry point definition

magdaaniol committed:
pin prodigy version

magdaaniol committed:
fix extension str comparison

ines committed:
Tidy up and add docstring

ines committed:
Update types

ines committed:
Update CLI shortcuts

ines committed:
Remove .docx for now

Can't create image-based previews easily

ines committed:
Increment version

ines committed:
Fix token offsets

ines committed:
Adjust color for consistency

ines committed:
Add meta to stream

magdaaniol commented on setup.cfg:
```diff
@@ -9,14 +9,17 @@ author_email = contact@explosion.ai
     pytesseract==0.3.10
     numpy<2.0.0
+    spacy-layout>=0.0.4
+    prodigy>=1.17.0
```
```suggestion
    prodigy>=1.17.1
```

magdaaniol commented on prodigy_pdf/spans.py:
```diff
@@ -0,0 +1,304 @@
+        focus=focus or [],
+    )
+    stream = Stream.from_iterable(layout_stream.get_stream())
+    if add_ents:
```
```suggestion
    if add_ents:
         labels = resolve_labels(nlp, "ner", recipe_labels=labels)
```
Otherwise if the user specifies --add-ents and does not provide labels (because they want all model labels) nothing gets added as labels is None.

magdaaniol commented on prodigy_pdf/spans.py:
```diff
@@ -0,0 +1,304 @@
+from typing import List, Optional, Tuple
+
+import pypdfium2 as pdfium
+from docling_core.types.doc.labels import DocItemLabel
```
```suggestion
from docling_core.types.doc.labels import DocItemLabel
from prodigy.components.preprocess import resolve_labels
```

ines committed:
Add pdf.spans.fetch

ines committed:
Apply suggestions from code review

Co-authored-by: Magdalena Aniol <96200718+magdaaniol@users.noreply.github.com>

magdaaniol commented on prodigy_pdf/spans.py:
```diff
@@ -302,3 +308,53 @@ def pdf_spans_manual(
+    source=Arg(help="Path to directory to load from"),
+    focus=Arg("--focus", "-f", help="Focus mode: annotate selected sections of a given type, e.g. 'text'"),
+    disable=Arg("--disable", "-d", help="Labels of layout spans to disable, e.g. 'footnote'"),
+    hide_preview=Arg("--hide-preview", "-HP", help="Hide side-by-side preview of layout"),
```
this should be irrelevant for a terminal only recipe, right?

ines replied:
It does impact the data it creates under the hood so at the moment, it's needed. But we could refactor it to make it not required

magdaaniol commented on prodigy_pdf/spans.py:
```diff
@@ -302,3 +308,53 @@ def pdf_spans_manual(
+    source: str,
+    focus: Optional[List[str]] = None,
+    disable: Optional[List[str]] = None,
+    hide_preview: bool = False,
```
ditto

ines committed:
Update recipe name

ines committed:
Merge branch 'feature/pdf-spans-manual' of https://github.com/explosion/prodigy-pdf into feature/pdf-spans-manual

magdaaniol committed:
fix indent

ines committed:
Only compute images if we need them

magdaaniol committed:
add answer field

magdaaniol committed:
set it to use python 3.10 and Prodigy 1.17.1

magdaaniol committed:
add answer field on top level

magdaaniol committed:
fix python version

ines committed:
Require fewer options for fetch recipe

Apply layout settings afterwards so fetched data can be reused more easily and make CSS the same for both interface types

ines committed:
Merge branch 'feature/pdf-spans-manual' of https://github.com/explosion/prodigy-pdf into feature/pdf-spans-manual

ines committed:
Tidy up
